### ISSUE TITLE
Embedding Sparse

### ISSUE THREAD

**Issue by wulidongdong** [pytorch/pytorch#17912]
## 🐛 Can not use sparse tensor to initial Embedding layer

 I want to create a sparse embedding layer, but I find out that it is not able to  use a sparse tensor to initial it.

![image](https://user-images.githubusercontent.com/15437641/54199132-1faa8c80-4503-11e9-9a71-559c401d4a8e.png)

it is not ok as well:
![image](https://user-images.githubusercontent.com/15437641/54199232-613b3780-4503-11e9-8ccc-69e7a47c20cd.png)

I try to use this way, it works, but the "to_dense()" function will cause too much memory occupied, that is not what  I want.
![image](https://user-images.githubusercontent.com/15437641/54199395-c42cce80-4503-11e9-8205-5a6150253e5f.png)

so is there any other way to initial a sparse embedding. Or other solutions to store a words co-occurrence matrix (means very large).

Thanks for your time!

**Comment by SsnL** [comment]
What does a sparse embedding matrix even mean? By definition, you have a vector for each word, then it ought to be dense. `sparse=True` in `nn.Embedding` makes the gradient sparse, not the weight matrix.

**Comment by mruberry** [comment]
Closing due to age and lack of clarity about the request. Please reopen addressing @SsnL if this is still an issue.
